### PR TITLE
Directional Vehicle Parts for mounted tools, seats, windshields, rams, and doors

### PR DESCRIPTION
**Pull request by mrkybe**
#### Summary

SUMMARY: Infrastructure "Adds a script for creating directional vehicle parts from json."

#### Purpose of change

A step towards making this a reality <https://github.com/CleverRaven/Cataclysm-DDA/issues/35861>

I applied the script to most vehicle parts: seats, rams, doors, tools, windshields.  Game will now ask you which variant you want to mount of that piece, like it does for frames.

#### Describe the solution

Edit the script to set the `parts.json` to be modified.  Possibly comment out some of the variants you don't want included.  The script takes every non-abstract part in `parts.json` and makes it an abstract one.  Then it creates a variant for each of those similar to the ones that exist for frames/boards.  For convenience it also writes a textfile with every newly created id, hopefully that can help tileset makers update their tilesets?

### CODE REVIEW

File: tools/make_vehicle_parts_directions.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return json_data
+   
+path = os.path.join(os.getcwd(), default_path)
+output_path = os.path.join(os.getcwd(), default_output)
+result = gen_new(path)
+print(*ids_to_create_directional_parts_for, sep = "\n")
+
+with open(output_path, "w") as jf:
+    json.dump(result, jf, ensure_ascii=False, sort_keys=True)
+    subprocess.Popen(['D:/Cataclysm-DDA/tools/format/json_formatter.exe', output_path], shell=True)
```
Comment (ifreund) [inline tools/make_vehicle_parts_directions.py:56]: This is only going to work on windows

**Comment by kevingranade** [comment]
> Unfortunately, any part that's been converted to abstract doesn't exist anymore. So vehicle definitions that were using that part will have an error instead.

If I understand you correctly, this is not ok, it's going to impose a huge rewrite and migration effort that is going to have to be adressed immediately.

Can you instead have it introduce a new abstract part and then create a non-abstrsct post that inherits from it and uses the original name?

I.e. some part "heavy_frame"
The script moves the contents of "heavy_frame" to "heavy_frame_abstract" and also creates a "heavy_frame" that is a shallow copy of "heavy_frame_abstract".

With that done everything can be migrated over piecemeal.

**Comment by mrkybe** [comment]
> Can you instead have it introduce a new abstract part and then create a non-abstrsct post that inherits from it and uses the original name?
>
> I.e. some part "heavy_frame"
> The script moves the contents of "heavy_frame" to "heavy_frame_abstract" and also creates a "heavy_frame" that is a shallow copy of "heavy_frame_abstract".
> 
> With that done everything can be migrated over piecemeal.

Perfect, that's a lot easier than my plan of writing a script to update the vehicle definitions.  I'll do that.

File: tools/make_vehicle_parts_directions.py (hunk 1 above)
Comment (mrkybe) [reply tools/make_vehicle_parts_directions.py]: The script I based this off, this https://github.com/CleverRaven/Cataclysm-DDA/blob/master/tools/json_tools/pocket_mags.py and it wasn't working for me on Windows.  I don't actually expect anyone to be running this script, as I'm committing the transformed JSON along side it.  It's more for documentation purposes.

**Comment by I-am-Erk** [comment]
How's it going with this?

**Comment by mrkybe** [comment]
> 
> 
> How's it going with this?

This is ready.

**Comment by I-am-Erk** [comment]
image
🎶Did I ever tell you you're my hero?

image
🎶You're everything I would like to beeeeee

image
🎶I can fly higher than an eeeagle

image
🎶Can you help me script changes to the default json vehicles now so they use directional paaarts pleeeeeease

**Pull request merged**